dataplane/saiserver/l2.go

```diff
@@ -63,71 +65,111 @@ func (mg *l2mcGroup) CreateL2McGroup(ctx context.Context, req *saipb.CreateL2McG
 	return &saipb.CreateL2McGroupResponse{Oid: id}, nil
 }
 
-func (mg *l2mcGroup) CreateL2McGroupMember(ctx context.Context, req *saipb.CreateL2McGroupMemberRequest) (*saipb.CreateL2McGroupMemberResponse, error) {
-	id := mg.mgr.NextID()
-	// Update table entry
-	r := fwdconfig.TableEntryAddRequest(mg.dataplane.ID(), L2MCGroupTable).AppendEntry(
-		fwdconfig.EntryDesc(fwdconfig.ExactEntry(fwdconfig.PacketFieldBytes(fwdpb.PacketFieldNum_PACKET_FIELD_NUM_L2MC_GROUP_ID).WithUint64(req.GetL2McGroupId())))).Build()
-	for _, p := range mg.groups {
-		r.Entries[0].Actions = append(r.Entries[0].Actions, &fwdpb.ActionDesc{
+// updateGroupMember updates the member of a L2MC group.
+// If m is nil, remove mid from the group(key: group id), otherwise add m to groups with mid as the key.
+func (mg *l2mcGroup) updateGroupMember(ctx context.Context, gid, mid uint64, m *l2mcGroupMember) error {
+	if mg.groups[gid] == nil {
+		return status.Errorf(codes.FailedPrecondition, "L2MC group %d does not exist", gid)
+	}
+	if m == nil {
```

Comment:
this can be greatly simplified
```go
if m == nil {
  delete(mg.groups[gid], mid)
} else {
  mg.groups[gid][mid] = m
}
mList := []L2MemberGroupMember{}
for _, m := mg.groups[gid] {
 mList = append(mList, {})
}
mgr.PopulateAttr()
```

Reply:
Done